packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -293,6 +309,46 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
         }
     }
 
+    /**
+     * For each line in the tree (this.state.timegraphTree), parse the metadata and try to find
+     * matches with the key / values pairs of the clicked event.
+     * It counts the amount of metadata matches and returns the TimeGraphEntry with the greatest amount,
+     * which is the most likely result.
+     *
+     * @params payload
+     *      Object with information about the line clicked in the Events Table
+     * @return
+     *      Correspondent TimeGraphEntry from this.state.timegraphTree
+     */
+    private findElement(payload: { [key: string]: string | { [key: string]: string }}): TimeGraphEntry | undefined {
+        let element: TimeGraphEntry | undefined = undefined;
+        let max = 0;
+        if (payload && payload.load) {
+            this.state.timegraphTree.forEach(el => {
+                if (el.metadata) {
+                    let cnt = 0;
+                    Object.entries(el.metadata).forEach(([key, value]) => {
+                        if (typeof (payload.load) !== 'string') {
+                            const val = payload.load[key];
+                            if (val !== undefined) {
+                                const num = Number(val);
+                                if ((num !== undefined && num === value[0]) || (val === value[0])) {
```

Comment:
I was debating about changing to a single value than array. In Trace Compass it's an array, because the metadata is part of a ICoreElementResolver which could have multiple values (e.g. selected CPUs). I wasn't sure if I would break any future use cases if I remove the array here. The events table values are strings not arrays. Not sure how to compare it in a loop then. Each value matches or one matches? @PatrickTasse Please let me know.

Reply:
I was thinking theoretically, you have an event with tid=456 and a time graph entry with metadata tid=[123, 456, 789], that would be one match. So that would add one loop level. But in practice it would always be an array of 1 so not much cost added for the loop.

Reply:
I'll do that. Thanks